src/main/java/com/nha/abdm/wrapper/hip/hrp/GatewayCallbackController.java

```diff
@@ -90,15 +93,22 @@ public void confirmCall(@RequestBody ConfirmResponse confirmResponse)
    * @param linkOnInitResponse Response from ABDM gateway after auth/init which has transactionId.
    */
   @PostMapping({"/v0.5/users/auth/on-init"})
-  public void onAuthInitCall(@RequestBody LinkOnInitResponse linkOnInitResponse) {
+  public ErrorResponse onAuthInitCall(@RequestBody LinkOnInitResponse linkOnInitResponse)
```

Comment:
As discussed, the return type could be `ResponseEntity<GatewayResponse>` and function can create this object accordingly.

Reply:
made changes, thanks